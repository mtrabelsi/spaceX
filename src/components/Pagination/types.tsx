import { USearchFilter } from "../../state/redux/types"

export type SetFilterType = (f : USearchFilter) => void

export type PaginationCPropType = {
    itemsCountPerPage: number,
    limit: number, 
    offset:number, 
    lastQueriedLength: number,
    leftLabel?: string,
    rightLabel?: string,
    setFilter: SetFilterType
}

export type PageChangeSignature = (n: number) => void