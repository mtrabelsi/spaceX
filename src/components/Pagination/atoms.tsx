import s from 'styled-components';
import { Button } from '../Button/atoms';

export const RightNavigator = s(Button)`
`;
export const LeftNavigator = s(Button)`
`;
export const PaginationWrapper = s.section`
    display: flex;
    justify-content: space-between;
    padding: 7px;
    flex-wrap: wrap;
`;
