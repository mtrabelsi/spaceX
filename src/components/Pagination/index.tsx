import React from 'react';
import {
  RightNavigator,
  LeftNavigator,
  PaginationWrapper,
} from './atoms';
import { PaginationCPropType, PageChangeSignature } from './types';

function PaginationC(props: PaginationCPropType) {
  const {
    limit,
    setFilter,
    lastQueriedLength,
    itemsCountPerPage,
    leftLabel,
    rightLabel,
    activePage,
    setActivePage,
  } = props;
  const handlePageChange : PageChangeSignature = (newPage) => {
    setFilter({
      limit,
      offset: newPage * limit,
    });
    setActivePage(newPage);
  };
  return (
    <PaginationWrapper>
      <LeftNavigator
        disabled={activePage === 0}
        onClick={() => handlePageChange(activePage - 1)}
      >
        {(leftLabel || 'Prev').toUpperCase()}
      </LeftNavigator>
      <RightNavigator
        onClick={() => handlePageChange(activePage + 1)}
        disabled={lastQueriedLength < itemsCountPerPage}
      >
        {(rightLabel || 'Next').toUpperCase()}
      </RightNavigator>
    </PaginationWrapper>
  );
}

export default PaginationC;
