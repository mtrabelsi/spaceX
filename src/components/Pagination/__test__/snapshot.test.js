import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from '..';

it('Should consider all props and generate unique html based on that', () => {
  const tree = renderer
    .create(
      <Pagination
        limit="sample title"
        setFilter={() => console.log('set filter')}
        setActivePage={() => console.log('set active page')}
        activePage={7}
        leftLabel="left label"
        rightLabel="right label"
        lastQueriedLength={3}
        itemsCountPerPage={5}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
