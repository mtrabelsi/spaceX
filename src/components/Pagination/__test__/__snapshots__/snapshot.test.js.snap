// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should consider all props and generate unique html based on that 1`] = `
<section
  className="sc-AxhUy beKucN"
>
  <button
    className="sc-AxjAm sc-AxhCb fVYgu"
    disabled={false}
    onClick={[Function]}
  >
    LEFT LABEL
  </button>
  <button
    className="sc-AxjAm sc-AxiKw dcLUtn"
    disabled={true}
    onClick={[Function]}
  >
    RIGHT LABEL
  </button>
</section>
`;
